addons/payment/tests/test_flows.py

```diff
@@ -347,3 +327,41 @@ def test_transaction_wrong_token(self):
         self.assertIn(
             "odoo.exceptions.ValidationError: The access token is invalid.",
             response.text)
+
+    @mute_logger('odoo.addons.payment.models.payment_transaction')
+    def _test_max_one_payment_request_from_portal(self, flow):
+
+        payment_option_id = self.create_token().id if flow == 'token' else self.acquirer.id
+        data = {
+            'amount': self.amount,
+            'currency_id': self.currency.id,
+            'partner_id': self.partner.id,
+            'access_token': None, flow: flow,
+            'payment_option_id': payment_option_id,
+            'reference_prefix': 'test',
+            'tokenization_requested': True,
+            'landing_route': 'Test',
+            'is_validation': False,
+            'invoice_id': self.invoice.id,
+            'flow': flow,
+        }
+
+        with unittest.mock.patch(
+                'odoo.addons.payment.utils.check_access_token', return_value=True
+        ), \
```

Comment:
Can't you provide a valid access token and avoid this patch? :thinking:

Reply:
`generate_access_token` uses `request.env`, calling it outside a request isn't advised. I could simulate it by recreating the function to set a real access token and hope we never change it... But isn't it a bit overkill when this simple patch is easy to understand?

Reply:
There is a test helper for that (cf `_generate_test_access_token`), not sure it will be appropriate here, but did you try? :D

Reply:
Nop, didn't know that! I'll look at it (as well as the test not working anymore since I rebase '^^)

Reply:
`resolved`